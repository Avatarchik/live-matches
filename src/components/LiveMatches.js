// @flow

import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'react-slick';
import LiveEvent from './LiveEvent';
import { SLIDER_SETTINGS } from '../constants';
import type { Score, LiveEvents } from './LiveEvent';

// Static type checking for Live Score
type LiveData = {
  score: Score
};

// Static type checking for Matches
type Match = {
  event: LiveEvents,
  liveData: LiveData
};

// Props type for LiveMatches component
type Props = {
  liveMatches: {
    length: number,
    liveEvents: []
  }
};

const LiveMatches = (props: Props) => {
  const { liveMatches } = props;
  return (
    <React.Fragment>
      <Slider {...SLIDER_SETTINGS}>
        {liveMatches.length !== 0
          ? liveMatches.liveEvents.map((match: Match) => {
              return (
                <div key={match.event.id}>
                  <LiveEvent
                    liveEvent={match.event}
                    liveScore={match.liveData}
                  />
                </div>
              );
            })
          : null}
      </Slider>
    </React.Fragment>
  );
};

LiveMatches.prototype = {
  liveMatches: PropTypes.shape({
    liveEvents: PropTypes.array
  })
};

export default LiveMatches;
