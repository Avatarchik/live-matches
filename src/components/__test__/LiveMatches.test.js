import React from 'react';
import { shallow } from 'enzyme';
import LiveMatches from '../LiveMatches';

const props = {
  liveMatches: {
    length: 0
  }
};

describe('Test suite for LiveMatches Component', () => {
  // create a UI snapshot to indentify UI changes or design regressions

  it('render correctly text component', () => {
    const component = shallow(<LiveMatches {...props} />);
    expect(component).toMatchSnapshot();
  });
});
