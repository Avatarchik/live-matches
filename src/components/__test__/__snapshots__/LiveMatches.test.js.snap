// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for LiveMatches Component render correctly text component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LiveMatches
    liveMatches={
      Object {
        "length": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Slider
        autoplay={true}
        autoplaySpeed={3000}
        className="live-match-slider"
        dots={false}
        slidesToScroll={1}
        slidesToShow={1}
        speed={500}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoplay": true,
        "autoplaySpeed": 3000,
        "children": null,
        "className": "live-match-slider",
        "dots": false,
        "slidesToScroll": 1,
        "slidesToShow": 1,
        "speed": 500,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Slider
          autoplay={true}
          autoplaySpeed={3000}
          className="live-match-slider"
          dots={false}
          slidesToScroll={1}
          slidesToShow={1}
          speed={500}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoplay": true,
          "autoplaySpeed": 3000,
          "children": null,
          "className": "live-match-slider",
          "dots": false,
          "slidesToScroll": 1,
          "slidesToShow": 1,
          "speed": 500,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
